test_vm/vendor/cookbooks/dbfit_test/metadata.rb

```diff
@@ -7,7 +7,7 @@
 version          '0.3.0'
 depends          'mysql2_chef_gem'
 depends          'yum-mysql-community'
-depends          'mysql', '~> 8.2'
+depends          'mysql'
```

Comment:
I'm leaning towards `>= 8.2` and falling back to `~> 8.2` if the mysql cookbook develops new issues in 9.x. 

Agreed; a separate PR would be better.